src/super_gradients/training/sg_trainer/sg_trainer.py

```diff
@@ -954,7 +954,13 @@ def forward(self, inputs, targets):
             training_params = dict()
         self.train_loader = train_loader or self.train_loader
         self.valid_loader = valid_loader or self.valid_loader
-        if len(self.train_loader.dataset) % self.train_loader.batch_size != 0 and not self.train_loader.drop_last:
+
+        if hasattr(self.train_loader, "batch_sampler") and self.train_loader.batch_sampler is not None:
```

Comment:
how is this change related?

Reply:
thats my and point here https://github.com/Deci-AI/super-gradients/pull/587#issue-1512612668 which I found only during running yolox recipe in the test (batch sampler is used -> batch_size=None, which lead this line to crash).
Could've taken it out and merge it before but it seemed such a small thing.